tools/expotools/src/Changelogs.ts

```diff
@@ -63,14 +79,107 @@ export class Changelog {
     this.filePath = filePath;
   }
 
+  /**
+   * Resolves to `true` if changelog file exists, `false` otherwise.
+   */
+  async fileExistsAsync(): Promise<boolean> {
+    return await fs.pathExists(this.filePath);
+  }
+
+  /**
+   * Lexifies changelog content and returns resulting tokens.
+   */
   async getTokensAsync(): Promise<Markdown.Tokens> {
     if (!this.tokens) {
-      await fs.access(this.filePath, fs.constants.R_OK);
-      this.tokens = Markdown.lexify(await fs.readFile(this.filePath, 'utf8'));
+      try {
+        await fs.access(this.filePath, fs.constants.R_OK);
+        this.tokens = Markdown.lexify(await fs.readFile(this.filePath, 'utf8'));
+      } catch (error) {
+        this.tokens = [];
+      }
     }
     return this.tokens;
   }
 
+  /**
+   * Reads versions headers, collects those versions and returns them.
+   */
+  async getVersionsAsync(): Promise<string[]> {
+    const tokens = await this.getTokensAsync();
+    const versionTokens = tokens.filter(
+      (token) => token.type === Markdown.TokenType.HEADING && token.depth === VERSION_HEADING_DEPTH
+    ) as Markdown.HeadingToken[];
+
+    return versionTokens.map((token) => token.text);
+  }
+
+  /**
+   * Returns the last version in changelog.
+   */
+  async getLastPublishedVersionAsync(): Promise<string | null> {
+    const versions = await this.getVersionsAsync();
+    return versions.find((version) => semver.valid(version)) ?? null;
+  }
+
+  /**
+   * Reads changes between two given versions and returns them in JS object format.
+   * If called without params, then only unpublished changes are returned.
+   */
+  async getChangesAsync(
+    fromVersion?: string,
+    toVersion: string = UNPUBLISHED_VERSION_NAME
+  ): Promise<ChangelogChanges> {
+    const tokens = await this.getTokensAsync();
+    const versions = {};
```

Comment:
```suggestion
    const versions = {};
```
If this variable had a more meaningful type it would be easier to read the main loop of this function 😉